ThePhasesOfCraftship/2_best_practice_first/testingBasics/exercises/2_Arrange_Act_Assert_Backwards/2_1_Password_Validator/src/index.spec.ts

```diff
@@ -1,9 +1,120 @@
+import { PasswordValidator, ValidationResult, ValidationError, ValidationErrorType } from ".";
+
+// Utility function for asserting successful validation
+function expectToPass(validationResult: ValidationResult) {
+  expect(validationResult.result).toBe(true);
+  expect(validationResult.errors).toHaveLength(0);
+}
+
+// Utility function for asserting failed validation
+function expectToFail(validationResult: ValidationResult, ...errors: ValidationError[]) {
+  expect(validationResult.result).toBe(false);
+  expect(validationResult.errors).toHaveLength(errors.length);
+  expect(validationResult.errors).toMatchObject(errors);
+}
 
 describe('password validator', () => {
 
-  test('hello', () => {
-    expect("between 5 and 15").toContain('5 and 15')
-  })
-})
+  it('Should list all the occurring validation errors', () => {
+    // arrange
+    const password = '';
+
+    // act
+    const validationResult = PasswordValidator.Validate(password);
+
+    // assert
```

Comment:
I think we don't need the comments for AAA

Reply:
I agree, I found them helpful at the first TDD exercise, but I totally agree that they are not needed once you know what you're supposed to do :)